packages/x-date-pickers-pro/src/internal/hooks/useMobileRangePicker/useMobileRangePicker.tsx

```diff
@@ -87,7 +87,7 @@ export const useMobileRangePicker = <
     externalSlotProps: componentsProps.field,
     additionalProps: {
       ...pickerFieldProps,
-      readOnly,
+      readOnly: readOnly ?? true,
```

Comment:
Fixes mobile range picker field editing (should not be possible—readOnly)

Reply:
Resolved it in a possibly [more correct way](https://github.com/mui/mui-x/pull/7135/files#r1049367534) using the result of your snippet.